lib/processEventDataHelper.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+const moment = require('moment-timezone');
+const _ = require('lodash');
+const Q = require('q');
+
+
+
+function processEventData(cfg, messageBody) {
+
+  console.log("Processing event data ...");
+  var defer = Q.defer();
+
+  function checkRequiredFields(cfg, messageBody) {
+
+    console.log("Checking required fields ...");
+    if (!messageBody.start) {
+      return defer.reject(new Error("Start Time missing! This field is required!"));
+    }
+    if (!messageBody.end) {
+      return defer.reject(new Error("End Time missing! This field is required!"));
+    }
+    if (!cfg.timeZone) {
+      return defer.reject(new Error("Time Zone missing! This field is required!"));
+    }
+    if (!cfg.calendarId) {
+      return defer.reject(new Error("Calendar ID missing! This field is required!"));
+    }
+    if ((cfg.bodyContentType) && (!messageBody.body)) {
+      return defer.reject(new Error("Body Type provided, but Body Content is missing!"));
+    }
+
+    return buildPostBody(cfg, messageBody);
+  }
+
+  function buildPostBody() {
+
+    function formatDate(date, timeZone, format) {
+      if (moment(date).isValid()) {
+        return moment.tz(date, timeZone).format(format);
+      }
+      if (moment(date, 'x').isValid()) {
+        return moment(date, 'x').tz(timeZone).format(format);
+      }
+      return defer.reject(new Error("Invalid date " + date));
+    }
+
+    let result = _.cloneDeep(messageBody);
+
+    if (cfg.importance) {
+      result.importance = cfg.importance;
+    }
+
+    if (cfg.sensitivity) {
+      result.sensitivity = cfg.sensitivity;
+    }
+
+    if (cfg.showAs) {
+      result.showAs = cfg.showAs;
+    };
+
+    if (cfg.bodyContentType) {
+      result.body.contentType = cfg.bodyContentType;
+    }
+
+    result.end.timeZone = cfg.timeZone;
+    result.start.timeZone = cfg.timeZone;
+
+    let FORMAT = 'YYYY-MM-DDTHH:mm:ss';
+    if ((cfg.isAllDay) && (cfg.isAllDay === 'true')) {
+      result.isAllDay = cfg.isAllDay;
+      // The last day given as parameter is exclusive.
+      // If the user wants to add all day event for 21-23 for ex,
+      // outlook expects something like: 21 (00:00 AM) - 24 (00:00 AM)
+      FORMAT = 'YYYY-MM-DD';
+      result.start.dateTime = formatDate(result.start.dateTime.trim(), cfg.timeZone, FORMAT);
+      result.end.dateTime = formatDate(result.end.dateTime.trim(), cfg.timeZone, FORMAT);
+      result.end.dateTime = moment(result.end.dateTime).add(1,'days').format(FORMAT);
+    } else {
+      result.start.dateTime = formatDate(result.start.dateTime.trim(), cfg.timeZone, FORMAT);
+      result.end.dateTime = formatDate(result.end.dateTime.trim(), cfg.timeZone, FORMAT);
+    }
+
+    defer.resolve(result);
+  }
+
+  checkRequiredFields(cfg, messageBody);
+  return defer.promise;
+}
+
+module.exports.processEvent = processEventData;
```

Comment:
It's a good practise to name the exported function the the real implementation same.